mesh_segmenter/src/mesh_segmenter.cpp

```diff
@@ -8,19 +8,17 @@
 #include <vtkDataArray.h>
 #include <vtkCellData.h>
 #include <vtkPointData.h>
-#include <vtkDoubleArray.h>
 #include <vtkFloatArray.h>
 
 #include <mesh_segmenter/mesh_segmenter.h>
 
 namespace mesh_segmenter
 {
-
 void MeshSegmenter::setInputMesh(vtkSmartPointer<vtkPolyData> mesh)
 {
   input_mesh_ = mesh;
 
-  if(!triangle_filter_)
+  if (!triangle_filter_)
```

Comment:
PCL has a region growing segmentation implementation. It appears you wrote your own using some curvature measure. I don't see anything wrong with what you wrote. I'm just surprised you didn't find a region growing segmention method already available in vtk.

Reply:
We discussed this in person, but I'll put it here for posterity.

This segmentation was here from prior work. That is why I am using it. However, it does seem to have some issues, so it might be worth trying the PCL region growing. The only issue there is that it operates in point clouds instead of meshes. Finally, Levi mentioned that the way we are doing this is very similar to the way his afront code in PCL works. He already calculates curvatures and has done the legwork for threaded operation - merging meshes segments that collide, etc. So that also seems like a very promising avenue. It would seem that the afront is a few if statements away from being a segmenter.

As for VTK region growing algorithm, I didn't see anything immediately. But I can't say I did a deep dive. I mostly saw this was already here and went with it.